### PR TITLE
Cleaned up SupportWarn

### PR DESCRIPTION
**Pull request by zt64**
* Stopped patching if the user has already agreed to both

### CODE REVIEW

**Comment by zt64** [comment]
<@!343383572805058560> merge when

File: Aliucord/src/main/java/com/aliucord/coreplugins/SupportWarn.kt [hunk 1]
```diff
@@ -85,7 +85,7 @@ internal class SupportWarn : Plugin(Manifest("SupportWarn")) {
                     dialog.dismiss()
                 }
 
-                dialog.show((it.thisObject as WidgetChatInput).parentFragmentManager, "Warning")
+                dialog.show((thisObject).parentFragmentManager, "Warning")
```
Comment (Vendicated) [inline Aliucord/src/main/java/com/aliucord/coreplugins/SupportWarn.kt:88]:
```suggestion
                dialog.show(thisObject.parentFragmentManager, "Warning")
```

File: Aliucord/src/main/java/com/aliucord/coreplugins/SupportWarn.kt [hunk 2]
```diff
@@ -7,71 +7,71 @@ import android.widget.*
...
-
-            val gateButtonText = binding.root.findViewById<TextView>(gateButtonTextId)
-            val chatWrap = binding.root.findViewById<LinearLayout>(chatWrapId)
-            val gateButtonImage = binding.root.findViewById<ImageView>(gateButtonImageId)
-            val gateButtonArrow = binding.root.findViewById<ImageView>(gateButtonArrowId)
-            val gateButtonLayout = binding.root.findViewById<RelativeLayout>(gateButtonLayoutId)
+            val thisObject = it.thisObject as WidgetChatInput
+            val root = WidgetChatInput.`access$getBinding$p`(thisObject).root
+            val chatWrap = root.findViewById<LinearLayout>(chatWrapId)
+            val gateButtonLayout = root.findViewById<RelativeLayout>(gateButtonLayoutId)
```
Comment (Vendicated) [inline Aliucord/src/main/java/com/aliucord/coreplugins/SupportWarn.kt:67]:
```suggestion
            val gateButtonLayout = root.findViewById<ViewGroup>(gateButtonLayoutId)
```

for future proofing

File: Aliucord/src/main/java/com/aliucord/coreplugins/SupportWarn.kt [hunk 3]
```diff
@@ -7,71 +7,71 @@ import android.widget.*
...
-        811262084968742932L, // #core-development
-        PLUGIN_REQUESTS_CHANNEL_ID,
-    )
-
-    private val chatWrapId = Utils.getResId("chat_input_wrap", "id")
-    private val gateButtonTextId = Utils.getResId("chat_input_member_verification_guard_text", "id")
-    private val gateButtonImageId = Utils.getResId("chat_input_member_verification_guard_icon", "id")
-    private val gateButtonArrowId = Utils.getResId("chat_input_member_verification_guard_action", "id")
-    private val gateButtonLayoutId = Utils.getResId("guard_member_verification", "id")
+    private val SettingsAPI.acceptedPrdNotRequests: Boolean by settings.delegate(false)
```
Comment (Vendicated) [inline Aliucord/src/main/java/com/aliucord/coreplugins/SupportWarn.kt:22]: why on settings instead of simply on class?

**Review by Vendicated — CHANGES_REQUESTED** [review]

File: Aliucord/src/main/java/com/aliucord/coreplugins/SupportWarn.kt (hunk 3 above)
Comment (zt64) [reply Aliucord/src/main/java/com/aliucord/coreplugins/SupportWarn.kt:22]: Prettier

**Review by Vendicated — APPROVED** [review]

**Pull request merged**